include/gamepad/controller.hpp

```diff
@@ -50,30 +178,79 @@ class Controller {
-         * @note Create a separate instance for each task.
+         *
+         * @b Example:
+         * @code {.cpp}
+         * while (true) {
+         *   Gamepad::master.update();
+         *   // do robot control stuff here...
+         *   pros::delay(25);
+         * }
+         * @endcode
+         *
          */
         void update();
         /**
          * Get the state of a button on the controller.
-         * @param button Which button's state you want.
+         * @param button Which button to return
+         *
+         * @b Example:
+         * @code {.cpp}
+         * if(Gamepad::master[DIGITAL_L1]) {
+         *   // do something here...
+         * }
+         * @endcode
+         *
          */
         const Button& operator[](pros::controller_digital_e_t button);
         /**
          * Get the value of a joystick axis on the controller.
-         * @param joystick Which joystick axis's value to return
+         * @param joystick Which joystick axis to return
+         *
+         * @b Example:
+         * @code {.cpp}
+         * // control a motor with a joystick
+         * intake.move(Gamepad::master[ANALOG_RIGHT_Y]);
+         * @endcode
+         *
          */
         float operator[](pros::controller_analog_e_t joystick);
-        TODO("hide memebrs and expose getters/const refs")
-        Button L1 {}, L2 {}, R1 {}, R2 {}, Up {}, Down {}, Left {}, Right {}, X {}, B {}, Y {}, A {};
-        float LeftX = 0, LeftY = 0, RightX = 0, RightY = 0;
+        const Button& L1 {m_L1};
+        const Button& L2 {m_L2};
+        const Button& R1 {m_R1};
+        const Button& R2 {m_R2};
+        const Button& Up {m_Up};
+        const Button& Down {m_Down};
+        const Button& Left {m_Left};
+        const Button& Right {m_Right};
+        const Button& X {m_X};
+        const Button& B {m_B};
+        const Button& Y {m_Y};
+        const Button& A {m_Down};
+        const float& LeftX = m_LeftX;
+        const float& LeftY = m_LeftY;
+        const float& RightX = m_RightX;
+        const float& RightY = m_RightY;
         /// The master controller, same as @ref Gamepad::master
         static Controller master;
         /// The partner controller, same as @ref Gamepad::partner
         static Controller partner;
     private:
-        explicit Controller(pros::controller_id_e_t id)
+        Controller(pros::controller_id_e_t id)
             : controller(id) {}
 
+        Button m_L1 {}, m_L2 {}, m_R1 {}, m_R2 {}, m_Up {}, m_Down {}, m_Left {}, m_Right {}, m_X {}, m_B {}, m_Y {},
+            m_A {};
+        float m_LeftX = 0, m_LeftY = 0, m_RightX = 0, m_RightY = 0;
+        Button Fake {};
+        /**
+         * @brief Gets a unique name for a listener that will not conflict with user listener names. IMPORTANT: when
+         * using the function, you must register the listener by directly calling add_listener on the EventHandler, do
+         * NOT use onPress/addListener,etc.
```

Comment:
use `@important` command

```suggestion
         * @brief Gets a unique name for a listener that will not conflict with user listener names.
         *
         * @important: when using the function, you must register the listener by
         * directly calling add_listener on the EventHandler, do NOT use onPress/addListener,etc.
```